src/tools/generate_workspace/src/main/java/com/google/devtools/build/workspace/GenerateWorkspace.java

```diff
@@ -82,16 +90,42 @@ private static void printUsage(OptionsParser parser) {
         OptionsParser.HelpVerbosity.LONG));
   }
 
-  private GenerateWorkspace(String outputDir) {
+  private GenerateWorkspace(String reqOutputDir, String reqBuildOutputDir, String reqWorkspaceOutputDir) {
     this.handler = new EventHandler();
     this.fileSystem = getFileSystem();
     this.resolver = new Resolver(handler, new DefaultModelResolver());
     this.inputs = Lists.newArrayList();
-    if (outputDir.isEmpty()) {
-      this.outputDir = fileSystem.getPath(Files.createTempDir().toString());
+
+    File buildFile;
+    File workspaceFile;
+    if (!reqBuildOutputDir.isEmpty() || !reqWorkspaceOutputDir.isEmpty()) {
```

Comment:
Use positive conditions when possible (e.g., if reqOutputDir.isEmpty()), especially since requestBuildOutputDir is non-empty if-and-only-if reqWorkspaceOutputDir is non-empty.